M/Makemsix/build_tarballs.jl

```diff
@@ -0,0 +1,159 @@
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Makemsix"
+version = v"1.7.241"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/microsoft/msix-packaging.git",
+              "efeb9dad695a200c2beaddcba54a52c8320bd135"),
+    DirectorySource(joinpath(@__DIR__, "patches"))
+]
+
+
+# Script that will adapt to each platform
+script = raw"""
+    cd $WORKSPACE/srcdir/msix-packaging
+
+    # Update C++ standard to 17
+    sed -i 's/set(CMAKE_CXX_STANDARD 14)/set(CMAKE_CXX_STANDARD 17)/' CMakeLists.txt
+    find . -name "CMakeLists.txt" -type f -exec sed -i 's/set(CMAKE_CXX_STANDARD 14)/set(CMAKE_CXX_STANDARD 17)/' {} \;
+    find . -name "CMakeLists.txt" -type f -exec sed -i 's/cmake_minimum_required(VERSION 3.29.0 FATAL_ERROR)/cmake_minimum_required(VERSION 3.21.7)/' {} \;
+
+    cp ../lib/CMakeLists.txt lib/CMakeLists.txt
+    cp ../src/msix/PAL/Signature/OpenSSL/SignatureValidator.cpp src/msix/PAL/Signature/OpenSSL/SignatureValidator.cpp 
+
+    mkdir .vs
+    cd .vs
+
+    # Set up linker and include paths
+    export LDFLAGS="-L${libdir}"
+    export CPPFLAGS="-I${includedir}"
+    
+    # Ensure pkg-config can find our dependencies
+    export PKG_CONFIG_PATH="${libdir}/pkgconfig:${prefix}/share/pkgconfig"
+
+    export OPENSSL_ROOT_DIR="${prefix}"
+    export OPENSSL_CRYPTO_LIBRARY=${libdir}/libcrypto.${dlext}
+    export OPENSSL_SSL_LIBRARY=${libdir}/libssl.${dlext}
+
+    # Platform-specific configuration
+    if [[ "${target}" == *"-apple-darwin"* ]]; then
+
+        # Set architecture-specific paths based on target
+        if [[ "${target}" == *"aarch64-apple-darwin"* ]]; then
+            ARCH="aarch64"
+            DARWIN_VER="20"
+            OSX_ARCH="arm64"
+        else
+            ARCH="x86_64"
+            DARWIN_VER="14"
+            OSX_ARCH="x86_64"
+        fi
+
+        SYSROOT="/opt/${ARCH}-apple-darwin${DARWIN_VER}/${ARCH}-apple-darwin${DARWIN_VER}/sys-root"
+        CF_FRAMEWORK="${SYSROOT}/System/Library/Frameworks/CoreFoundation.framework"
+        FOUNDATION_FRAMEWORK="${SYSROOT}/System/Library/Frameworks/Foundation.framework"
+
+        cmake \
+          -DCMAKE_BUILD_TYPE=MinSizeRel \
+          -DSKIP_BUNDLES=on \
+          -DUSE_VALIDATION_PARSER=on \
+          -DUSE_MSIX_SDK_ZLIB=on \
+          -DCMAKE_OSX_ARCHITECTURES=${OSX_ARCH} \
+          -DMSIX_PACK=on \
+          -DMSIX_SAMPLES=on \
+          -DMSIX_TESTS=off \
+          -DMACOS=on \
+          -DCMAKE_OSX_SYSROOT=${SYSROOT} \
+          -DCOREFOUNDATION_LIBRARY=${CF_FRAMEWORK} \
+          -DFOUNDATION_LIBRARY=${FOUNDATION_FRAMEWORK} \
+          -DCMAKE_SHARED_LINKER_FLAGS="-Wl,-install_name,@rpath/libmsix.dylib" \
+          -DCMAKE_SHARED_LIBRARY_SUFFIX=".dylib" \
+          -DXML_PARSER=xerces \
+          -DCRYPTO_LIB=openssl \
+          -DCMAKE_PREFIX_PATH=${prefix} \
+          -DBUILD_SHARED_LIBS=ON \
+          ..
```

Comment:
It'd be better to rationalise this in a way such that all common code is set once.  Something like
```sh
CMAKE_FLAGS=(
...
)
if [[ something ]]; then
    CMAKE_FLAGS+=(...)
elif [[ something else ]];
    CMAKE_FLAGS+=(...)
fi
cmake "${CMAKE_FLAGS[@]}" ..
```
Code duplication is bad and error-prone.

Reply:
I will take a look at how I can improve it.